runtime/interpreter/value_range.go

```diff
@@ -0,0 +1,260 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package interpreter
+
+import (
+	"fmt"
+
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/errors"
+	"github.com/onflow/cadence/runtime/sema"
+)
+
+// NewInclusiveRangeValue constructs an InclusiveRange value with the provided start, end with default value of step.
+func NewInclusiveRangeValue(
+	interpreter *Interpreter,
+	locationRange LocationRange,
+	start IntegerValue,
+	end IntegerValue,
+	rangeType InclusiveRangeStaticType,
+) *CompositeValue {
+	startComparable, startOk := start.(ComparableValue)
+	endInclusiveComparable, endInclusiveOk := end.(ComparableValue)
+	if !startOk || !endInclusiveOk {
+		panic(errors.NewUnreachableError())
+	}
+
+	step := GetValueForIntegerType(1, rangeType.ElementType)
```

Comment:
Given that the `GetValueForIntegerType` calls are all with fixed integers, and the set of integer types is known and small, maybe we can lazily pre-compute a lookup table?

Reply:
Nice idea. Done, added it as a lookup table in Interpreter. Do let me know if you think it should be stored somewhere else.